sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/UserIdContext.java

```diff
@@ -153,21 +148,37 @@ public synchronized String getUserId() {
      *
      * @param userId user identifier.
      */
-    public synchronized void setUserId(String userId) {
-        mUserId = userId;
+    public void setUserId(String userId) {
+        if (isUserIdChanged(userId)) {
+            return;
+        }
 
-        // TODO if mUserId != userId
-        // TODO move out from synchronized
         /* Call listeners so that they can react on new token. */
         for (Listener listener : mListeners) {
             listener.onNewUserId(mUserId);
         }
     }
 
+    /**
+     * Check user identifier.
+     *
+     * @param userId user identifier.
+     * @return true if equals.
+     */
+    private synchronized boolean isUserIdChanged(String userId) {
```

Comment:
Changing private field is not reflected anywhere in description or name; 
Also, won't it be better to implement it like this?
```Java
return userId.equals(mUserId)
``` 
Note the order of fields (userId<>mUserId)